bin/network-monitor/src/deploy/account.rs

```diff
@@ -317,13 +271,11 @@ impl DataStore for MonitorDataStore {
         vault_root: Word,
         vault_key: Word,
     ) -> Result<AssetWitness, DataStoreError> {
-        let account = self.wallet_account.lock().await;
-
-        let account = if account_id == account.id() {
-            account.to_owned()
-        } else {
-            self.counter_account.lock().await.to_owned()
-        };
+        let account =
+            self.accounts.get(&account_id).cloned().ok_or_else(|| DataStoreError::Other {
+                error_msg: "unknown account".into(),
+                source: None,
+            })?;
```

Comment:
nit: This is repeated 3x
```rust
DataStoreError::Other {
                error_msg: "unknown account".into(),
                source: None,
            })?;
```

Might be worth adding a fn for it instead (`get_account(&self)` or so)